File: hanoi.cxx
#include <cstdio>

void hanoi(int n, char x, char y, char z) {
    if (n == 1) {
        printf("%d from %c to %c\n", n, x, z);
        return;
    } else {
        hanoi(n - 1, x, z, y);
        printf("%d from %c to %c\n", n, x, z);
        hanoi(n - 1, y, x, z);
    }
}

int main() {
    int n;
    n = 3;
    hanoi(n, 'A', 'B', 'C');

    return 0;
}

File: hanoi.go
package main

import "fmt"

func hanoi(n int, x string, y string, z string) {
    if n == 1 {
        fmt.Printf("%d from %s to %s\n", n, x, z)
        return
    } else {
        hanoi(n - 1, x, z, y);
        fmt.Printf("%d from %s to %s\n", n, x, z)
        hanoi(n - 1, y, x, z);
    }
}

func main() {
    n := 3
    hanoi(n, "A", "B", "C")
}
